apps/ex_nvr/lib/ex_nvr/model/recording.ex

```diff
@@ -58,6 +58,14 @@ defmodule ExNVR.Model.Recording do
     from(r in query, where: r.device_id == ^device_id and r.filename == ^name)
   end
 
+  def oldest_recordings_by_device(query \\ __MODULE__, device_id, limit) do
+    from(r in query,
+      where: r.device_id == ^device_id,
+      order_by: [desc: r.start_date],
```

Comment:
Are you sure it's `desc`, this way you'll delete the newest recordings not the oldest.